specification/marketplacecatalog/data-plane/Search/preview/2022-08-17-preview/search.json

```diff
@@ -519,6 +519,10 @@
               "x-ms-error-code": {
                 "type": "string",
                 "description": "Error code"
+              },
+              "azure-deprecating": {
+                "type": "string",
+                "description": "Deprecation Message"
               }
```

Comment:
Could you confirm that your backend indeed return `azure-deprecating` header with text in runtime, like https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#deprecating-behavior-notification ?

Reply:
yes, should be in prod by end of this week (we are rolling a nw version with the new header).

"https://euap.catalogapi.azure.com/search?gallery=AppSourceApps&select=UniqueProductId&top=20&Language=en&Market=ALL&api-version=2022-09-25-preview"

the response contains header "azure-deprecating"with value set to:

"API versions 2022-09-25-preview & 2022-08-17-preview will retire on Jan. 31, 2024. After this date, this API version will no longer be supported. To continue using our services, it is essential to migrate to the new Search API preview version 2023-01-01.For detailed guidance, please refer to the updated API documentation, available at: https://learn.microsoft.com/en-us/rest/api/marketplacecatalog/2023-01-01-preview/search/get?tabs=HTTP If you have any questions or require assistance during the transition, please contact us at MKPL_Platform_API_DL@microsoft.com"

Reply:
The format seems not same as the guideline (especially the date).
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#deprecating-behavior-notification

Also please make sure you've met the requirement there (breaking change review and azure update).

@JeffreyRichter @mikekistler for awareness on breaking changes.
- marketplacecatalog deprecates 2022-08-17-preview and 2022-09-25-preview.
- Latest preview is 2023-01-01-preview. It is more than 2 month old (so I assume it meets the policy, of 90 days overlap window).

Reply:
Hi @mikekistler,
The communication is ready and about to be sent (our Product Manager is working with Shashank Vineet on it).
I added the header 'azure-deprecating' with wrong formatted message as @weidongxu-microsoft mention, but understood this header is not mandatory and dont have to match the communication (the same message is about to be sent in communication).
the 'wrong formatted' header should be available in prod in the next 2 days.
I can change the "azure-deprecating" value we return to match the format you sent, but it will reach prod only next week. 
please let me know how to proceed, we would like to complete this PR as soon as possible (it contains a red banner on top letting users on the deprecation, although it is not required)

Reply:
@mikekistler Let us know if you have opinion here.

I assume `azure-deprecating` header does not need to be an exact copy of your communication (via email or other) to customer.

`azure-deprecating` header in Swagger and runtime is intended to notify user that calling the API, giving them context that the plan/schedule/time of the deprecation.

Reply:
Hi @mikekistler @weidongxu-microsoft 
Thanks for your help.
Decided with @weidongxu-microsoft  that we will complete this PR and publish the documentation as is, and next week "azure-depreacting" message will be fixed.
since we dont have the url part of the message, the only change will be in the date format (**in bold**).

wrong formatted message:
"API versions 2022-09-25-preview & 2022-08-17-preview will retire on **Jan. 31, 2024**. After this date, this API version will no longer be supported. To continue using our services, it is essential to migrate to the new Search API preview version 2023-01-01.For detailed guidance, please refer to the updated API documentation, available at: https://learn.microsoft.com/en-us/rest/api/marketplacecatalog/2023-01-01-preview/search/get?tabs=HTTP If you have any questions or require assistance during the transition, please contact us at [MKPL_Platform_API_DL@microsoft.com](mailto:MKPL_Platform_API_DL@microsoft.com)"

new message:
"API versions 2022-09-25-preview & 2022-08-17-preview will retire on **2024-01-31**. After this date, this API version will no longer be supported. To continue using our services, it is essential to migrate to the new Search API preview version 2023-01-01. For detailed guidance, please refer to the updated API documentation, available at: https://learn.microsoft.com/en-us/rest/api/marketplacecatalog/2023-01-01-preview/search/get?tabs=HTTP If you have any questions or require assistance during the transition, please contact us at [MKPL_Platform_API_DL@microsoft.com](mailto:MKPL_Platform_API_DL@microsoft.com)"

@weidongxu-microsoft  please confirm the new message, so I can start rolling the change.